src/v/cloud_storage/remote.cc

```diff
@@ -1299,36 +1300,27 @@ ss::future<upload_result> remote::delete_objects_sequentially(
             .node = std::make_unique<retry_chain_node>(&parent)};
       });
 
-    auto results = co_await ss::do_with(
-      bucket,
-      std::move(key_nodes),
-      std::vector<upload_result>{},
-      [this, ctxlog](auto& bucket, auto& key_nodes, auto& results) {
-          results.reserve(key_nodes.size());
-          return ss::max_concurrent_for_each(
-                   key_nodes.begin(),
-                   key_nodes.end(),
-                   concurrency(),
-                   [this, &bucket, &results, ctxlog](auto& kn) -> ss::future<> {
-                       vlog(ctxlog.trace, "Deleting key {}", kn.key);
-                       return delete_object(bucket, kn.key, *kn.node)
-                         .then([&results](auto result) {
-                             results.push_back(result);
-                         });
-                   })
-            .handle_exception_type([ctxlog](const std::exception& ex) {
```

Comment:
+1 to switching to coroutine. Just so I understand, this fix could also have been to change `.handle_exception_type([ctxlog](const std::exception& ex)` to `.handle_exception([ctxlog](std::exception_ptr eptr)`?

Reply:
Ha 🤦, yea I think so